packages/@tinacms/mdx/src/parse/parseShortcode.test.ts

```diff
@@ -0,0 +1,87 @@
+/**
+
+Copyright 2021 Forestry.io Holdings, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+*/
+import { it, expect, describe } from 'vitest'
+import { parseShortcode } from './parseShortcode'
+
+describe('parseShortcode', () => {
+  describe('with keyed field', () => {
+    it('parses attributes', () => {
+      const result = parseShortcode('{{< signature foo="bar123">}}', {
+        name: 'signature',
+        label: 'Signature',
+        match: {
+          start: '{{<',
+          end: '>}}',
+        },
+        fields: [
+          {
+            name: 'foo',
+            label: 'foo label',
+            type: 'string',
+          },
+        ],
+      })
+      expect(result).toEqual('<signature foo="bar123">\n</signature>')
+    })
+  })
+
+  describe('with unkeyed attributes', () => {
+    it('parses attributes', () => {
+      const result = parseShortcode('{{< signature "bar123" >}}', {
+        name: 'signature',
+        label: 'Signature',
+        match: {
+          start: '{{<',
+          end: '>}}',
+        },
+        fields: [
+          {
+            name: 'text',
+            label: 'text',
+            type: 'string',
+          },
+        ],
+      })
+      expect(result).toEqual('<signature text="bar123">\n</signature>')
```

Comment:
Won't this break some user's shortcodes? I would expect that when it's unkeyed going _in_ it remains unkeyed going _out_

Reply:
I believe this is more or less doing what it did before for unkeyed strings. parseShortcode translates `<% foobar "123" %>` to `<foobar text="123"` just so that it is usable by our rich-text field.

Then when it's output back to .md, stringifyShortcode translates it back to:
`<% foobar "123" %>`

This only happens when a "text" field is defined on a template. That special logic is kind of why I'd like to make that field name something less common (like _values)